agent/agent.go

```diff
@@ -210,10 +227,12 @@ type agent struct {
 	// done is a channel used for cleanup.
 	done chan struct{}
 
+	// TODO: do we need separate clocks here?
```

Comment:
That's exactly the question I have reading this 😄maybe they were introduced for tests? I would take a look and get rid of extra clocks if there's no real need for it.